### PR TITLE
Use NativePointer instead of Long as a cross-platfrom counterpart of Long

### PR DESCRIPTION
**Pull request by Schahen**
This is a minimal implementation and most likely I missed something but I suggest to use it as a base for all following improvements in this direction

### CODE REVIEW

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/Bitmap.kt [hunk 1]
```diff
@@ -613,7 +618,7 @@ class Bitmap internal constructor(ptr: Long) : Managed(ptr, _FinalizerHolder.PTR
      * @param rowBytes  size of pixel row or larger; may be zero
      * @return          true if pixel storage is allocated
      */
-    fun allocPixels(info: ImageInfo, rowBytes: Long): Boolean {
```
Comment (alexander-gorshenev) [inline skiko/src/commonMain/kotlin/org/jetbrains/skia/Bitmap.kt:616]: rowBytes, наверное, не указатель.
На jni-ной стороне там написано `jlong rowBytes`.
И в конечном итоге превращается в `size_t rowBytes`

Comment (Schahen) [reply skiko/src/commonMain/kotlin/org/jetbrains/skia/Bitmap.kt:616]: а вот это как раз те младенцы, которых я вылил с водой!

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/impl/Native.kt [hunk 2]
```diff
@@ -1,13 +1,59 @@
...
+import org.jetbrains.skia.FilterMipmap
+import org.jetbrains.skia.IPoint
+import org.jetbrains.skia.IRange
+import org.jetbrains.skia.ImageInfo
 
-    companion object {
-        fun getPtr(n: Native?): Long
-    }
+expect class NativePointer
+expect val NULLPNTR: NativePointer
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skia/impl/Native.kt:10]: Does `expect const val` work?

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/impl/Native.kt [hunk 3]
```diff
@@ -1,13 +1,59 @@
...
 
+
+expect fun toIPoint(p: NativePointer): IPoint
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skia/impl/Native.kt:13]: None of those belongs to this file.

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/impl/Native.kt [hunk 4]
```diff
@@ -1,13 +1,59 @@
...
+
+/**
+ * Returns true if rowBytes is valid for this ImageInfo.
+ *
+ * @param rowBytes  size of pixel row including padding
+ * @return          true if rowBytes is large enough to contain pixel row and is properly aligned
+ */
+expect fun ImageInfo.isRowBytesValid(rowBytes: NativePointer): Boolean
+
+expect fun Int.toNativePointer(): NativePointer
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skia/impl/Native.kt:46]: Not sure if this function makes sense anywhere but on JS, and even there it seems to be a bug.

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/Data.kt [hunk 5]
```diff
@@ -3,17 +3,21 @@ package org.jetbrains.skia
...
 import kotlin.jvm.JvmStatic
 
 /**
  * Data holds an immutable data buffer.
  */
-class Data internal constructor(ptr: Long) : Managed(ptr, _FinalizerHolder.PTR) {
+class Data internal constructor(ptr: NativePointer) : Managed(ptr, _FinalizerHolder.PTR) {
     companion object {
-        fun makeFromBytes(bytes: ByteArray, offset: Long = 0, length: Long = bytes.size.toLong()): Data {
+        fun makeFromBytes(bytes: ByteArray, offset: NativePointer = NULLPNTR, length: NativePointer = bytes.size.toNativePointer()): Data {
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skia/Data.kt:20]: shall be Long or changed to Int, not sure if we care about > 4G arrays.

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/Data.kt [hunk 6]
```diff
@@ -38,48 +42,48 @@ class Data internal constructor(ptr: Long) : Managed(ptr, _FinalizerHolder.PTR)
...
-        external fun _nMakeEmpty(): Long
+        external fun _nMakeEmpty(): NativePointer
 
         init {
             staticLoad()
         }
     }
 
-    val size: Long
+    val size: NativePointer
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skia/Data.kt:76]: Long or Int.

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/Bitmap.kt [hunk 7]
```diff
@@ -6,9 +6,14 @@ import org.jetbrains.skia.impl.Managed
 import org.jetbrains.skia.impl.Stats
 import org.jetbrains.skia.impl.reachabilityBarrier
 import org.jetbrains.skia.ExternalSymbolName
+import org.jetbrains.skia.impl.NULLPNTR
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skia/Bitmap.kt:9]: NULLPNTR -> maybe Native.NullPointer, i.e.
class Native {
   companion object {
      val NullPointer: NativePointer
    }
}```

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/ImageInfo.kt [hunk 8]
```diff
@@ -256,4 +254,35 @@ class ImageInfo(val colorInfo: ColorInfo, val width: Int, val height: Int) {
...
-}
\ No newline at end of file
+}
+
+/**
+ * Returns minimum bytes per row, computed from pixel getWidth() and ColorType, which
+ * specifies getBytesPerPixel(). Bitmap maximum value for row bytes must fit
+ * in 31 bits.
+ */
+expect val ImageInfo.minRowBytes: NativePointer
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skia/ImageInfo.kt:264]: Why it is a pointer at all? Looks like an Int.

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/ImageInfo.kt [hunk 9]
```diff
@@ -256,4 +254,35 @@ class ImageInfo(val colorInfo: ColorInfo, val width: Int, val height: Int) {
...
+ *
+ *
+ * Returns zero if height is zero.
+ *
+ * @param rowBytes  size of pixel row or larger
+ * @return          memory required by pixel buffer
+ *
+ * @see [https://fiddle.skia.org/c/@ImageInfo_computeByteSize](https://fiddle.skia.org/c/@ImageInfo_computeByteSize)
+ */
+expect fun ImageInfo.computeByteSize(rowBytes: NativePointer): NativePointer
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skia/ImageInfo.kt:279]: Same, whole file looks redundant.

**Pull request merged**